pandas/tests/frame/test_to_csv.py

```diff
@@ -1108,3 +1108,13 @@ def test_to_csv_with_dst_transitions(self):
             df.to_pickle(path)
             result = pd.read_pickle(path)
             assert_frame_equal(result, df)
+
+    def test_round_trip_scientific_no_exponent(self):
+        df = DataFrame({'x': [2.5], 'y': [42], 'z': ['2E']})
```

Comment:
this needs to go in io/test_parsers.py

and use self.read_csv

as this test under all engines (this might fail under other parsers)